static/js/stream_popover.js

```diff
@@ -486,8 +486,10 @@ function build_move_topic_to_stream_popover(e, current_stream_id, topic_name) {
             default_text: $t({defaultMessage: "No streams"}),
             include_current_item: false,
             value: current_stream_id,
+            on_update: allow_or_disallow_submission,
```

Comment:
It seems likely that we need to call this function when updating the topic input, not just the stream input, since it's a function of both.

Reply:
Yikes, good catch. Will fix.